core/src/main/java/com/amplifyframework/core/category/Category.java

```diff
@@ -167,35 +162,56 @@ public final P getPlugin(@NonNull final String pluginKey) {
      */
     @NonNull
     public final Set<P> getPlugins() {
-        return new HashSet<>(plugins.values());
+        return Immutable.of(new HashSet<>(plugins.values()));
     }
 
     /**
      * Obtain the registered plugin for this category.
      * @return The only registered plugin for this category
      * @throws IllegalStateException
      *         If the category is not configured, or if there are no
-     *         plugins associated to the category
+     *         plugins associated to the category, or if Amplify has not
+     *         been configured
      */
     @NonNull
-    protected final P getSelectedPlugin() {
-        if (!isConfigured) {
-            throw new IllegalStateException("This category is not yet configured." +
-                    "Make sure you added it with Amplify.addPlugin and then called Amplify.config");
+    protected final P getSelectedPlugin() throws IllegalStateException {
+        try {
+            validateConfigurationState(ConfigurationState.CONFIGURED);
+        } catch (AmplifyException amplifyException) {
+            // Every category behavior calls getSelectedPlugin(), so we can't realistically
+            // ask each one to catch a (checked) AmplifyException.
+            throw new IllegalStateException(amplifyException);
         }
+
         if (plugins.isEmpty()) {
             throw new IllegalStateException(
-                    "Tried to get a plugin but that plugin was not present." +
-                    "Check if the plugin was added originally or perhaps was already removed."
+                "Tried to get a plugin but that plugin was not present. " +
+                "Check if the plugin was added originally or perhaps was already removed."
             );
-        }
-        if (plugins.size() > 1) {
+        } else if (plugins.size() > 1) {
             throw new IllegalStateException(
-                    "Tried to get a default plugin but there are more than one to choose from in this category." +
-                    "Call getPlugin(pluginKey) to choose the specific plugin you want to use in this case."
+                "Tried to get a default plugin but there are more than one to choose from in this category. " +
+                "Call getPlugin(pluginKey) to choose the specific plugin you want to use in this case."
             );
         }
 
         return getPlugins().iterator().next();
     }
+
+    private void validateConfigurationState(ConfigurationState targetState) throws AmplifyException {
+        synchronized (configurationState) {
+            if (!configurationState.get().equals(targetState)) {
+                throw new AmplifyException(
+                    "Amplify was already configured.",
+                    "Be sure to only call Amplify.configure once"
+                );
+            }
+        }
+    }
```

Comment:
We discussed a bug about this :)

Should have `validateConfigured()` and `throwIfConfigured()` separately to clearly indicate two validation cases

Reply:
Fixing! Thanks for the catch.